### PR TITLE
Role Improvements

### PR DESCRIPTION
**Pull request by ghsolomon**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Review by cnrudd — APPROVED** [review]
Reviewed and ran locally.  All good!

File: admin/tabs/general/roles/RoleModel.ts [hunk 1]
```diff
@@ -16,6 +16,13 @@ import {HoistRole, RoleMemberType, RoleServiceConfig} from './Types';
 export class RoleModel extends HoistModel {
     static PERSIST_WITH = {localStorageKey: 'xhAdminRolesState'};
 
+    static fmtDirectoryGroup(name?: string): string {
```
Comment (lbwexler) [inline admin/tabs/general/roles/RoleModel.ts:19]: don't think name is optional here

Comment (ghsolomon) [reply admin/tabs/general/roles/RoleModel.ts:19]: Thanks - actually should be `string | null`, but I'll change to just `string`

File: admin/tabs/general/roles/details/members/RoleMembersModel.ts [hunk 2]
```diff
@@ -175,13 +187,39 @@ export class RoleMembersModel extends HoistModel {
                 {
                     field: 'name',
                     autosizeMaxWidth: 300,
-                    renderer: (name, {record}) =>
-                        record.get('type') === types.DIRECTORY_GROUP
-                            ? this.fmtDirectoryGroup(name)
-                            : name,
-                    tooltip: true
+                    renderer: (name, {record}) => {
```
Comment (lbwexler) [inline admin/tabs/general/roles/details/members/RoleMembersModel.ts:190]: might help put this grid config on a diet if we were to define this renderer and the renderer below as functions at the bottom of the file.

File: admin/tabs/general/roles/editor/form/RoleFormModel.ts [hunk 3]
```diff
@@ -226,4 +263,22 @@ export class RoleFormModel extends HoistModel {
             debounce: 250
         };
     }
+
+    private async lookupDirectoryGroupAsync(directoryGroup: string, recordId: string) {
+        try {
+            const {data} = await XH.fetchJson({
+                autoAbortKey: recordId,
```
Comment (lbwexler) [inline admin/tabs/general/roles/editor/form/RoleFormModel.ts:270]: nice to see this used -- was surprised looking at the FetchService that this abortKey is not actually specific to the the url!  Wondering if the best practice here is to include some more context in this key

**Review by lbwexler — APPROVED** [review]

**Pull request merged**